src/test/java/com/puppycrawl/tools/checkstyle/bdd/InlineConfigParser.java

```diff
@@ -37,6 +37,10 @@
     /** A pattern matching the symbol: "/". */
     private static final Pattern SLASH_PATTERN = Pattern.compile("[\\\\/]");
 
+    /** A pattern to find the string: "// violation". */
+    private static final Pattern VIOLATION_PATTERN = Pattern
+            .compile(".*//\\s*violation\\s*$");
```

Comment:
@timurt trailing whitespaces don't make sense since we already enforce check for this.
Should we change this to ```".*//\\s*violation.*$"```?

Reply:
`.*//\\s*violation.*$` means anything after violation, this comment will match `// violationblablabla` 

if there will be no trailing whitespaces, you can use `.*//\\s*violation$`

Reply:
Then we'll have to refactor these cases?
https://github.com/checkstyle/checkstyle/blob/4ef9ff2fa6ad2002fdbaad0b767afb6b03cdf581/src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/javadoctagcontinuationindentation/InputJavadocTagContinuationIndentationBlockTag.java#L84

Reply:
I think it'll be good to refactor is we have to support this https://github.com/checkstyle/checkstyle/pull/10182#discussion_r659023717 in future.

Reply:
okay, then lets match `violation` comment, one or more spaces followed by any character or empty line

smth like `.*//\\s*violation[$|\s+.*$]`

Reply:
> one or more spaces followed by any character

but ```      * <ul> // violation, not correctly indented ``` does not have one or more spaces after violation

Reply:
okay) `.*//\\s*violation[$|\W+.*$]`

Reply:
done